core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportClusterAllocationExplainAction.java

```diff
@@ -248,11 +252,12 @@ public static ClusterAllocationExplanation explainShard(ShardRouting shard, Rout
             Float weight = weights.get(node);
             IndicesShardStoresResponse.StoreStatus storeStatus = nodeToStatus.get(node);
             NodeExplanation nodeExplanation = calculateNodeExplanation(shard, indexMetaData, node, decision, weight,
-                    storeStatus, shard.currentNodeId(), indexMetaData.activeAllocationIds(shard.getId()));
+                    storeStatus, shard.currentNodeId(), indexMetaData.activeAllocationIds(shard.getId()),
+                    allocation.hasPendingAsyncFetch());
             explanations.put(node, nodeExplanation);
         }
-        return new ClusterAllocationExplanation(shard.shardId(), shard.primary(),
-                shard.currentNodeId(), remainingDelayMillis, ui, explanations);
+        return new ClusterAllocationExplanation(shard.shardId(), shard.primary(), shard.currentNodeId(),
+                remainingDelayMillis, ui, allocation.hasPendingAsyncFetch(), explanations);
```

Comment:
`allocation.hasPendingAsyncFetch()` will always return false here. The field that is used to determine this value is set by Primary/ReplicaShardAllocator. Even if this field were correctly set here, it would still be the wrong value to determine whether the shard can be allocated or not. The primary/replica shard allocator is only interested in knowing whether there are still pending fetches for the targeted shard id.